### PR TITLE
Add DB repositories to access cluster and identity_cluster tables

### PR DESCRIPTION
**Pull request by alexeykazakov**
TODO:
- [x] Tests

Part of https://github.com/fabric8-services/fabric8-cluster/issues/15

### CODE REVIEW

**Comment by alexeykazakov** [comment]
Ready for review.

File: cluster/repository/identity_cluster.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+// Create creates a new record.
+func (m *GormIdentityClusterRepository) Create(ctx context.Context, c *IdentityCluster) error {
+	defer goa.MeasureSince([]string{"goa", "db", "identity_cluster", "create"}, time.Now())
+	err := m.db.Create(c).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_id":  c.ClusterID.String(),
+			"identity_id": c.IdentityID.String(),
+			"err":         err,
+		}, "unable to create the cluster")
```
Comment (dipak-pawar) [inline cluster/repository/identity_cluster.go]: how about `"unable to create identity cluster"`?

File: cluster/repository/identity_cluster.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	c := IdentityCluster{IdentityID: identityID, ClusterID: clusterID}
+
+	result := m.db.Delete(&c)
+
+	if result.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_id":  c.ClusterID.String(),
+			"identity_id": c.IdentityID.String(),
+			"err":         result.Error,
+		}, "unable to delete the cluster")
```
Comment (dipak-pawar) [inline cluster/repository/identity_cluster.go]: how about `"unable to delete the identity cluster"`?

File: cluster/repository/identity_cluster_blackbox_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+func assertContainsCluster(t *testing.T, clusters []repository.Cluster, cluster *repository.Cluster) {
+	require.NotEqual(t, uuid.UUID{}, cluster.ClusterID)
+	for _, cls := range clusters {
+		if cls.ClusterID == cluster.ClusterID {
+			return
+		}
+	}
+	assert.Fail(t, "didn't find cluster")
```
Comment (dipak-pawar) [inline cluster/repository/identity_cluster_blackbox_test.go]: How about giving some hint about `clusterID` in failure message?

File: application/repository/base/exister.go [hunk 4]
```diff
@@ -63,7 +63,7 @@ func CheckExistsWithCustomIDColumn(ctx context.Context, db *gorm.DB, tableName,
 
 // CheckExists does the same as Exists for a hard deletable item but only returns the error value; thereby
 // being a handy convenience function.
-func CheckHardDeletableExists(ctx context.Context, db *gorm.DB, tableName string, id string) error {
-	_, err := exists(ctx, db, tableName, "id", id, false)
+func CheckHardDeletableExists(ctx context.Context, db *gorm.DB, tableName string, idColumnName, id string) error {
```
[the remark this answers is not in the document]
Comment (alexeykazakov) [reply application/repository/base/exister.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/c53acb82e55793ca31aa2cb944e9333f53881ae7

File: application/repository/base/exister.go [hunk 5]
```diff
@@ -63,7 +63,7 @@ func CheckExistsWithCustomIDColumn(ctx context.Context, db *gorm.DB, tableName,
 
 // CheckExists does the same as Exists for a hard deletable item but only returns the error value; thereby
```
[the remark this answers is not in the document]
Comment (alexeykazakov) [reply application/repository/base/exister.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/c53acb82e55793ca31aa2cb944e9333f53881ae7

File: cluster/repository/identity_cluster.go (hunk 1 above)
Comment (alexeykazakov) [reply cluster/repository/identity_cluster.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/c53acb82e55793ca31aa2cb944e9333f53881ae7

File: cluster/repository/identity_cluster.go (hunk 2 above)
Comment (alexeykazakov) [reply cluster/repository/identity_cluster.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/c53acb82e55793ca31aa2cb944e9333f53881ae7

File: cluster/repository/identity_cluster_blackbox_test.go (hunk 3 above)
Comment (alexeykazakov) [reply cluster/repository/identity_cluster_blackbox_test.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/e81043f2775af1cd77a47a8671ff304baf49e83c

**Comment by alexeykazakov** [comment]
@dipak-pawar I've addressed your comments.

**Comment by alexeykazakov** [comment]
thanks @dipak-pawar !

**Pull request merged**